### ISSUE TITLE
Schema $refs are not pointing correctly

### ISSUE THREAD

**Issue by bpred754** [jlalmes/trpc-openapi#37]
Hi there,

Thank you for taking the time to put this library together! I just came across an issue when opening the swagger UI - I see errors like so:

```
Could not resolve reference: #/properties/data/properties/id
```

From looking through the source, it looks like the `zod-to-json-schema` refStrategy (root is the default) is not mapping correctly. Maybe because the entire object is not being passed into the function?

**Comment by jlalmes** [comment]
Hi @bpred754, thanks for opening an issue! Could you share a code sample that will generate one of these invalid schemas - I can't seem to reproduce this.

**Comment by jlalmes** [comment]
Reproduction here: https://github.com/modtree/modtree/pull/358#issuecomment-1195478283

**Comment by jlalmes** [comment]
Hi @weiseng18. Can you confirm this is fixed in the new `trpc-openapi@0.5.0` release?

**Comment by StefanTerdell** [comment]
@jlalmes Hello! Just stumbled over this old issue and [this](https://github.com/modtree/modtree/pull/358) 

Just FYI [the basePath option](https://github.com/stefanTerdell/zod-to-json-schema#options-object) was made for this type of scenario. for the modtree example you could use this

```
zodToJsonSchema(zodSchema, { 
  target: 'openApi3',
  basePath: ["paths", "post", "responses", "200", "content", "application\\/json", "schema"]
})
```

Of course the basepath should be dynamically generated but yeah
